### DISCUSSION TITLE
V1 to V2.3.3 - forum link?

### DISCUSSION THREAD

**Discussion by NECPWAweb** [e107inc/e107#5200] [Q&A Updating]
So have done a test upgrade on my site and some things don't appear to work e.g. the link to the 'upgraded' pre existing forums /e107_plugins/forum/forum.php shows a blank page. Anyone have any idea why?

Tried a different theme but that made no difference. 

Checking the database: forum_t table looks full with the old posts as does forum table with the forum names but forum_post and forum_thread are empty?

**Comment by Jimmi08** [comment]
I skip all my complaints about not updating directly to 2.3,3 if you use the forum plugin.  They say it is safe, but I don't think so. 

So:
- check your error log
- set displayed PHP errors on
- set developer mode on
- use PHP 7 only

Reasons {I suppose you run forum update functionality successfully):
- not unique sef-url (2 of your forums/posts have the same title
- you are using an outdated plugin related to the forum (APOS, newsforum_menu, etc)
- LAN issue
- you are not using SEF-URLs for forum - your URL should be site/forum/  not legacy one.
and other reasons.

The blank page is easy to debug. Just put echo "I am here"; die;  in forum.php and you will how far you will get...

**Reply by NECPWAweb** [reply]
OK thanks using php 7 as 8 breaks my theme....
1st error:
<html><body>

Possible issue:  Missing a Unique Tablerender ID. Use $ns->setUniqueId() in the plugin  script prior to setContent(). See 'source code' for more information. | e_render->setContent()() | /home/tp9l2zckwdsp/public_html/e107_handlers/e_render_class.php | 391
-- | -- | -- | --



</body>
</html>
**2nd:**
 Use of undefined constant ADLAN_113 - assumed 'ADLAN_113' (this will throw an Error in a future version of PHP), Line 5 of /home/tp9l2zckwdsp/public_html/e107_plugins/forum/e_status.php

**Reply by Jimmi08** [reply]
Possible issue: Missing a Unique Tablerender ID. Use $ns->setUniqueId()  - it is not error, it is message for theme developer. 

Use of undefined constant ADLAN_113 - assumed 'ADLAN_113'  - this will break your site in PHP 8, but not in PHP 7 (at least I am not aware about it)
Look for fatal errors, not warnings.

**Reply by NECPWAweb** [reply]
Thanks, **forum plugin error log** shows this error:

[26-Feb-2024 14:47:59 UTC] PHP Fatal error:  Uncaught Error: Call to a member function parse_scbatch() on null in /home/tp9l2zckwdsp/public_html/e107_plugins/forum/forum_shortcodes.php:4
Stack trace:
#0 /home/tp9l2zckwdsp/public_html/e107_handlers/shortcode_handler.php(469): require_once()
#1 /home/tp9l2zckwdsp/public_html/e107_handlers/e107_class.php(1568): e_parse_shortcode->getScObject('forum_shortcode...', 'forum', NULL)
#2 /home/tp9l2zckwdsp/public_html/e107_plugins/forum/forum.php(121): e107::getScBatch('forum', true)
#3 /home/tp9l2zckwdsp/public_html/e107_plugins/forum/forum.php(35): forum_front->init(Object(e107forum))
#4 /home/tp9l2zckwdsp/public_html/e107_plugins/forum/forum.php(578): forum_front->__construct(Object(e107forum))
#5 {main}
  thrown in /home/tp9l2zckwdsp/public_html/e107_plugins/forum/forum_shortcodes.php on line 4

Now there isn't a file: e107_plugins/forum/forum_shortcodes.php - there is one, but it's in e107_plugins/forum/shortcodes/batch/forum_shortcodes.php

**PHP Errors shows:**
			
Notice Uninitialized string offset: 0, Line 141 of /home/tp9l2zckwdsp/public_html/e107_core/shortcodes/batch/navigation_shortcodes.php
--
Notice Uninitialized string offset: 0, Line 141 of /home/tp9l2zckwdsp/public_html/e107_core/shortcodes/batch/navigation_shortcodes.php
Notice Uninitialized string offset: 0, Line 141 of /home/tp9l2zckwdsp/public_html/e107_core/shortcodes/batch/navigation_shortcodes.php

**Reply by NECPWAweb** [reply]
OK so adding the 'echo' in forum.php it runs right to the end until just before line 575. putting the echo further down and nothing shows....

echo "I am here"; die; 
require_once(e_PLUGIN . 'forum/forum_class.php');
/** @var e107forum $forum */
$forum = new e107forum;

new forum_front($forum);

**Reply by Jimmi08** [reply]
forum/forum_shortcodes.php was a file in 1.0.4.  If you forget it in the forum folder, it is loaded...  clean cache, rescan plugins folder... (Tools/Database/Rescan plugins)
My advice, delete the forum folder and replace it with clean 2.3.3 forum plugin.   
If you did it already, then... I don't know, I don't use forum anymore.

**Reply by NECPWAweb** [reply]
Yes I did a clean install of the plugin and have spent the past few hours deleting all the old v1 stuff and re running the forum upgrade process

I don't actually use it as a 'forum'. It's a 'notice board' for events and errors with the club magazine etc. with no user registration or posting. There are only 2 of us who can post on site.

Now I have just switched themes to 'Voux' and the forums will load so it looks like it's theme related after all. I will try Bootstrap 5 tomorrow, but have had enough today.

Thanks for the input.

**Reply by tgtje** [reply]
Just in case, but think you checked it in your theme anyway : no files or folder used for forum? eg used for override the core..
Not sure at what date it started appearing in themes..

**Reply by NECPWAweb** [reply]
Yes the theme did have a 'forum' folder in it to overide the core so that was probably the issue, the Theme is 11 years old!

Only real issue I have now is attaching images in forum posts but will leave that to another day :) 

Test site is now running Bootstrap 5 @  https://necpwa.website/ that I'm slowly setting up as the custom page layouts all need revising along with the navigation.

**Reply by Jimmi08** [reply]
About attachments - there are some issues already (look for closed too).
